src/app.ts

```diff
@@ -122,7 +132,18 @@ async function app() {
   writeResources('profiles', outPackage.profiles);
   writeResources('extensions', outPackage.extensions);
   writeResources('vocabulary', [...outPackage.valueSets, ...outPackage.codeSystems]);
-  writeResources('examples', outPackage.instances);
+
+  // Sort instances into appropriate directories
+  const instances = cloneDeep(outPackage.instances);
```

Comment:
Yes. Since the filter functions mutate the object passed in, I had to clone it so that when it is used later on to get the total count of instances, the object had the right length. It also seemed safer in case we wanted to use the original `outPackage.instances` later in this function. I can add a comment explaining that if it will help. There is a comment notating the mutation, but it is in the util file, so it would probably help to have a comment here too. If you think the `filterXX` functions should not mutate at all, I can try to think of a different solution to this. However, this was the cleanest I had thought of at the time.

Reply:
No I'm all for `filterXX`, but also all for a comment.